rllib/agents/dqn/dqn_policy.py

```diff
@@ -362,12 +276,19 @@ def setup_late_mixins(policy, obs_space, action_space, config):
     TargetNetworkMixin.__init__(policy, obs_space, action_space, config)
 
 
-def _compute_q_values(policy, model, obs, obs_space, action_space):
+def _compute_q_values(policy, model, obs, explore):
     config = policy.config
+
+    policy.exploration.before_forward_pass(model, obs, explore=explore)
```

Comment:
This is an example for the before/forward/after split in a "manual" fashion (by providing a forward_fn in `build_tf_policy`).

Reply:
`forward_fn` replaces both `action_sampler_fn` AND `log_likelihood_fn`.